### DISCUSSION TITLE
How to find the crop geometry from an original and its cropped result

### DISCUSSION THREAD

**Discussion by halloleo** [ImageMagick/ImageMagick#3069] [Help] [answered]
In a GUI image editor I had cropped an image, so that I have the original image and the cropped result image.

Now I would like to use ImageMagick to crop a series of similar original images the same way. For this I  guess I need to know the _geometry_ argument for the `-crop` operator to do the crop.

How can I find out this geometry from the first original image and its cropped result image?

**Comment by fmw42** [comment]
When you crop the first image do not add +repace after -crop and save as PNG. Then

See the Origin geometry: argument values in identify -verbose your_image to get the offsets and the Geometry: argument for the WxH.

or

convert your_image_cropped.png -format "%wx%h%O" info:

**Reply by halloleo** [reply]
@fmw42 As @snibgo guessed I have only the image pixel data (because I cropped in _another_ app, an GUI photo editor).

**Comment by snibgo** [comment, marked as answer]
I think the OP has two images, where one is an unknown crop of the other. The task is to find what "-crop" would give the same result.

We can do a subimage-search of the small image within the large image:
```
magick compare -metric RMSE -subimage-search -dissimilarity-threshold 1 -similarity-threshold 0 large.png small.png NULL:
```
The text result will be something like:
```
8936.06 (0.136356) @ 36,12
```
So the (X,Y) offset of the small in the large is (36,12). So the crop will be "-crop WxH+36+12" where W and H are the dimensions of the small image.

For large images, this is painfully slow. For a one-off job, just go for a cup of coffee. There are faster but more complex methods if you need to do this often.

**Reply by halloleo** [reply]
Cool! Thx. I'll certainly try that!

**Reply by halloleo** [reply]
@snibgo Works as advertised! Execution time for a couple of images is in my case (JPEGs sized approx 1000x2000 resp. 800x1500) not an issue at all. Thanks again.
